wormhole_chain/x/tokenbridge/types/asset.go

```diff
@@ -0,0 +1,158 @@
+package types
+
+import (
+	"encoding/hex"
+	"fmt"
+	math "math"
+	"math/big"
+	"strconv"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	btypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	whtypes "github.com/wormhole-foundation/wormhole-chain/x/wormhole/types"
+)
+
+// Truncate an amount
+func Truncate(coin sdk.Coin, meta btypes.Metadata) (normalized sdk.Coin, err error) {
+	factor, err := truncFactor(meta)
+	if err != nil {
+		return normalized, err
+	}
+
+	amt := new(big.Int).Div(coin.Amount.BigInt(), factor)
+	return sdk.NewCoin(coin.Denom, sdk.NewIntFromBigInt(amt)), nil
+}
+
+// Untruncate an amount
+func Untruncate(coin sdk.Coin, meta btypes.Metadata) (normalized sdk.Coin, err error) {
+	factor, err := truncFactor(meta)
+	if err != nil {
+		return normalized, err
+	}
+
+	amt := new(big.Int).Mul(coin.Amount.BigInt(), factor)
+	return sdk.NewCoin(coin.Denom, sdk.NewIntFromBigInt(amt)), nil
+}
+
+// Compute truncation factor for a given token meta.
+// This is max(1, exponent - 8). If you divide an amount by this number, the
+// result will have 8 decimals.
+func truncFactor(meta btypes.Metadata) (factor *big.Int, err error) {
+	// Find the display denom to figure out decimals
+	var displayDenom *btypes.DenomUnit
+	for _, denom := range meta.DenomUnits {
+		if denom.Denom == meta.Display {
+			displayDenom = denom
+			break
+		}
+	}
+	if displayDenom == nil {
+		return new(big.Int), ErrDisplayUnitNotFound
+	}
+
+	if displayDenom.Exponent > math.MaxUint8 {
+		return nil, ErrExponentTooLarge
+	}
+
+	if displayDenom.Exponent > 8 {
+		return new(big.Int).SetInt64(int64(math.Pow10(int(displayDenom.Exponent - 8)))), nil
+	} else {
+		return big.NewInt(1), nil
+	}
+}
+
+var uwormChain uint16 = 1
+var uwormAddress = [32]byte{0x16, 0x58, 0x09, 0x73, 0x92, 0x40, 0xa0, 0xac, 0x03, 0xb9, 0x84, 0x40, 0xfe, 0x89, 0x85, 0x54, 0x8e, 0x3a, 0xa6, 0x83, 0xcd, 0x0d, 0x4d, 0x9d, 0xf5, 0xb5, 0x65, 0x96, 0x69, 0xfa, 0xa3, 0x01}
+
+func IsWORMToken(tokenChain uint16, tokenAddress [32]byte) bool {
+	// TODO(csongor): figure out WORM token address on Solana
+	// TODO(csongor): this should be configurable in the genesis
```

Comment:
I think we should add the wormtoken origin info into the genesis config instead

Reply:
Sounds good to me but I'd like to do this in a follow-up change.

Reply:
Actually, since we're going to be dropping this entire module maybe this isn't necessary.